goas/methods/nullifier/src/main.rs

```diff
@@ -0,0 +1,57 @@
+/// Nullifier Proof
+///
+/// Our goal: prove the nullifier nf was derived from a note that had previously been committed to.
+///
+/// More formally, nullifier statement says:
+/// for public input `nf` (nullifier) and `root_cm` (root of merkle tree over commitment set).
+/// the prover has knowledge of `output = (note, nf_pk, nonce)`, `nf` and `path` s.t. that the following constraints hold
```

Comment:
don't you have to prove knowledge of nullifier sk for *input* notes? why is this called output?

Reply:
Inputs are outputs of some other tx. Note commitments on chain are commitments to transaction outputs (recall UTXO stands for unspent transaction outputs.)

So here the prover is demonstrating that this input is derived from a valid output of some past transaction